src/interfaces.ts

```diff
@@ -131,18 +144,20 @@ export interface EffectsInfo {
 export interface HandlerInfo {
   lastProcessedBlockNumber: number
   lastProcessedBlockHash: string
+  lastIrreversibleBlockNumber: number
   handlerVersionName: string
-  effectRunMode: EffectRunMode
-  numberOfRunningEffects: number
-  effectErrors: string[]
+  isReplay?: boolean
+  effectRunMode?: EffectRunMode
+  numberOfRunningEffects?: number
+  effectErrors?: string[]
 }
 
 export interface ReaderInfo {
   currentBlockNumber: number
   startAtBlock: number
   headBlockNumber: number
-  onlyIrreversible: boolean
-  lastIrreversibleBlockNumber: number
+  onlyIrreversible?: boolean
```

Comment:
Is this really nullable? AbstractActionHandler uses false as a default, it would make sense that this property needs to be defined all the time.

Reply:
The `AbstractActionHandler` uses these, but it is conceivable to create an `ActionReader` (not from an extension of the `AbstractActionHandler`) that doesn't require this property (or `lastIrreversibleBlockNumber` below).